src/librustdoc/clean/utils.rs

```diff
@@ -574,3 +456,15 @@ crate fn find_nearest_parent_module(tcx: TyCtxt<'_>, def_id: DefId) -> Option<De
         None
     }
 }
+
+crate fn has_doc_flag(attrs: Attributes<'_>, flag: Symbol) -> bool {
+    for attr in attrs {
+        if attr.has_name(sym::doc) {
+            match attr.meta_item_list() {
+                Some(l) if list_contains_name(&l, flag) => return true,
+                _ => continue,
+            }
+        }
+    }
+    false
```

Comment:
Would be a bit shorter as:
```suggestion
    attrs.iter().any(|attr| {
        attr.has_name(sym::doc)
            && attr.meta_item_list().map_or(false, |l| list_contains_name(&l, flag))
    })
```

Reply:
Indeed!